File: mvvm/src/main/java/com/lookballs/mvvm/core/ILifecycleObserver.java
package com.lookballs.mvvm.core;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.OnLifecycleEvent;

/**
 * 创建人：lucas
 * 创建时间：2021/5/15 11:06
 * 类描述：activity生命周期监听基类
 */
public interface ILifecycleObserver extends LifecycleObserver {
    @OnLifecycleEvent(Lifecycle.Event.ON_ANY)
    void onLifecycleChanged(LifecycleOwner owner, Lifecycle.Event event);
}

File: mvvm-kt/src/main/java/com/lookballs/mvvm/core/ILifecycleObserver.kt
package com.lookballs.mvvm.core

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.OnLifecycleEvent

/**
 * 创建人：lucas
 * 创建时间：2021/5/15 11:06
 * 类描述：activity生命周期监听基类
 */
interface ILifecycleObserver : LifecycleObserver {
    @OnLifecycleEvent(Lifecycle.Event.ON_ANY)
    fun onLifecycleChanged(owner: LifecycleOwner, event: Lifecycle.Event)
}